integration/pages/BasePage.js

```diff
@@ -0,0 +1,52 @@
+const config = require('../utils/config.js');
+const { until } = require('selenium-webdriver');
+
+class BasePage {
+  constructor(driver) {
+    this.driver = driver;
+  }
+
+  async visit(route) {
+    await this.driver.get(config.baseUrl + route);
+  }
+
+  find(locator) {
+    return this.driver.findElement(locator);
+  }
+
+  async findAll(locator) {
+    return await this.driver.findElements(locator);
+  }
+
+  async click(locator) {
+    await this.find(locator).click();
```

Comment:
what if the locator does not exist? Do we have to wait for the locator before clicking?

Reply:
The duty of this function is just to click on the element. It is used in the scenario of clicking an element that already exists. This means that the user has to make sure that the element exists before using this method.

From my opinion, the duty you mentioned here is actually the `clickAfterElementAppears`. The flow of this scenario is:
1. The element does not exist at the beginning.
2. The element appears later.
3. Click the element.

for step 1, 2 we can use below method to check:

```js
  async isDisplayed(locator, timeout) {
    if (timeout) {
      await this.driver.wait(until.elementLocated(locator), timeout);
      await this.driver.wait(until.elementIsVisible(this.find(locator)), timeout);
      return true;
    } else {
      try {
        return await this.find(locator).isDisplayed();
      } catch (error) {
        return false;
      }
    }
  }
```

Reply:
This method doesn't seem necessary, its only purpose is to reduce the amount of code, but the functionality it provides is not complete and has risk to cause trouble.

I will remove `click()` and `type()` functions.

Reply:
>what if the locator does not exist? Do we have to wait for the locator before clicking?

There will be an `error.NoSuchEementError`.

Reply:
We should probably implement some wait here - people will use this click function without waiting for the element to be there. Better safe than sorry, right?

Reply:
You are right. I think we should avoid this by design, rather than dealing with it in function logic. We will have other `WebElement` related function in the future, e.g., `select()`, `check()`, ... , it's apparently wrong to handle this kind of wait situation inside each of them.

For now I will delete `click()`, `type` and probably the `isDisplayed()` function and try to find an alternative way later.

Reply:
Deleted `click()` and `type()` funtion. Refactored `isDisplayed` function a little bit. 
And set the default `timeout` to 5000 for all wait until functions.

```js
  // Check if the element is able to be found
  async isFound(locator, timeout = 5000) {
    return await this.driver.wait(until.elementLocated(locator), timeout);
  }

  // Check if the element is `displayed`
  async isDisplayed(locator, timeout = 5000) {
    return await this.driver.wait(until.elementIsVisible(this.find(locator)), timeout);
  }
```

So to check if an element appears and displayed, for example:
```js
  async checkIfTestAppHasLoaded() {
    const isFound = await this.isFound(ELEMENTS.appHeader);
    assert(isFound, 'The header of Test App was not found, Test App was not loaded successfully');

    const isDisplayed = await this.isDisplayed(ELEMENTS.appHeader);
    assert(isDisplayed, 'The header of Test App was not displayed, Test App was not loaded successfully');
  }
```